pkg/api/controller.go

```diff
@@ -160,21 +165,48 @@ func (c *Controller) DeleteObjects(w http.ResponseWriter, r *http.Request, body
 	writeResponse(w, http.StatusOK, response)
 }
 
-func (c *Controller) Logout(w http.ResponseWriter, _ *http.Request) {
-	http.SetCookie(w, &http.Cookie{
-		Name:     JWTCookieName,
-		Value:    "",
-		Domain:   c.Config.GetCookieDomain(),
-		Path:     "/",
-		HttpOnly: true,
-		Expires:  time.Unix(0, 0),
-		SameSite: http.SameSiteStrictMode,
-	})
-
+func (c *Controller) Logout(w http.ResponseWriter, r *http.Request) {
+	err := clearSession(w, r, c.sessionStore, InternalAuthSessionName)
+	if err != nil {
+		writeError(w, http.StatusInternalServerError, err)
+		return
+	}
+	err = clearSession(w, r, c.sessionStore, OIDCAuthSessionName)
+	if err != nil {
+		writeError(w, http.StatusInternalServerError, err)
+		return
+	}
 	writeResponse(w, http.StatusOK, nil)
 }
 
+// OauthCallback gets a code generated by an OIDC provider.
+// It exchanges the code for an id token, and saves the claims from the ID token on a session.
+func (c *Controller) OauthCallback(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	session, _ := c.sessionStore.Get(r, OIDCAuthSessionName)
+	if r.URL.Query().Get("state") != session.Values[StateSessionKey] {
+		writeError(w, http.StatusBadRequest, "Invalid state parameter.")
+		return
+	}
+	idTokenClaims, err := c.oidcAuthenticator.GetIDTokenClaims(ctx, r.URL.Query().Get("code"))
+	if err != nil {
+		writeError(w, http.StatusInternalServerError, err.Error())
+		return
+	}
+	session.Values[IDTokenClaimsSessionKey] = idTokenClaims
+	err = session.Save(r, w)
+	if err != nil {
+		writeError(w, http.StatusInternalServerError, err.Error())
+		return
+	}
+	http.Redirect(w, r, "/", http.StatusTemporaryRedirect)
+}
+
 func (c *Controller) Login(w http.ResponseWriter, r *http.Request, body LoginJSONRequestBody) {
+	err := clearSession(w, r, c.sessionStore, OIDCAuthSessionName)
```

Comment:
need to clear the old one - not the new one while login

Reply:
synced f2f: clearing the OIDC session here, and setting the "internal login" one.